common/src/chain/block/block_v1.rs

```diff
@@ -13,21 +13,25 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-use crate::chain::block::Block;
-use crate::chain::block::ConsensusData;
-use crate::chain::transaction::Transaction;
-use crate::primitives::id;
-use crate::primitives::id::Idable;
-use crate::primitives::{Id, H256};
-
 use serialization::{Decode, Encode};
 
-use super::block_header::BlockHeader;
-use super::timestamp::BlockTimestamp;
+use crate::{
+    chain::{
+        block::{Block, BlockReward, BlockRewardTransactable, ConsensusData},
+        transaction::Transaction,
+    },
+    primitives::{
+        id::{self, Idable},
+        Id, H256,
+    },
+};
+
+use super::{block_header::BlockHeader, timestamp::BlockTimestamp};
 
 #[derive(Debug, Clone, PartialEq, Eq, Encode, Decode, serialization::Tagged)]
 pub struct BlockV1 {
     pub(super) header: BlockHeader,
+    pub(super) reward: BlockReward,
     pub(super) transactions: Vec<Transaction>,
```

Comment:
Would it make sense to introduce a `BlockBody` type that couples together reward and transactions?